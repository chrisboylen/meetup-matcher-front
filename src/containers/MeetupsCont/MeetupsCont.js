import React, { Component } from 'react';
import { Meetup } from '../../components/Meetup/Meetup';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';

export class MeetupsCont extends Component {
  render() {
    const { meetups } = this.props;
    const meetupList = meetups.map((meetup, i) => (
      <Meetup 
        {...meetup}
        key={meetup.name+i}
      />
    ));

    return (
      <section className="meetups-cont">
        {meetupList}
      </section>
    );
  }
}

MeetupsCont.propTypes = {
  meetups: PropTypes.array,
  dispatch: PropTypes.func
};

export const mapStateToProps = ({ meetups }) => ({ meetups });

export default connect(mapStateToProps, null)(MeetupsCont);