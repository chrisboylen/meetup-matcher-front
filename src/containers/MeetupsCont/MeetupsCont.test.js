import React from 'react';
import { shallow } from 'enzyme';
import { MeetupsCont } from './MeetupsCont';

describe('MeetupsCont', () => {
  it('should match snapshot', () => {
    const meetups = [{},{}]
    const wrapper = shallow(<MeetupsCont meetups={meetups}/>);

    expect(wrapper).toMatchSnapshot();
  });
});