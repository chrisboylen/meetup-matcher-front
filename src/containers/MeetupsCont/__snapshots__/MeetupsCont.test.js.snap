// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetupsCont should match snapshot 1`] = `
<section
  className="meetups-cont"
>
  <Meetup
    key="NaN"
  />
  <Meetup
    key="NaN"
  />
</section>
`;
